aws_service/renewing-address.go

```diff
@@ -55,44 +55,77 @@ func (r *RenewingAddressGrpcServer) Listen() {
 
 // renewing the elastic ip using local ip and instanceId
 func (r *RenewingAddressGrpcServer) RenewElasticIp(context context.Context, reachedRequestLimit *pb.RenewingElasticIp) (*pb.RenewedElasticResult, error) {
-	ec2Address, err := r.getElasticPublicAddresses(reachedRequestLimit.InstanceId, reachedRequestLimit.PodIp)
 
 	log.Println("PodIp: ", reachedRequestLimit.PodIp)
 	log.Println("InstanceId: ", reachedRequestLimit.InstanceId)
+	var networkInterfaceId string
+	ec2Address, err := r.getElasticPublicAddresses(reachedRequestLimit.InstanceId, reachedRequestLimit.PodIp)
+
 	if err != nil {
 		log.Println(err)
-		return &pb.RenewedElasticResult{IsRenewed: false}, err
-	}
+		networkInterfaceId, err = r.getNetworkInterfaceId(reachedRequestLimit.InstanceId, reachedRequestLimit.PodIp)
 
-	err = r.disassociateAddress(*ec2Address.PublicIp)
-	if err != nil {
-		log.Println("disassociateAddress Error: ", err)
-		return nil, err
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		networkInterfaceId = *ec2Address.NetworkInterfaceId
+		err = r.disassociateAddress(*ec2Address.PublicIp)
+		if err != nil {
+			log.Println("disassociateAddress Error: ", err)
+			return nil, err
+		}
+
+		err = r.releaseElasticAddresses(*ec2Address.AllocationId)
+		if err != nil {
+			log.Println("Error ReleaseElasticAddress: ", err)
+			return nil, err
```

Comment:
I would wrape the error with `fmt.Errorf` and return it, since the worker will print it anyway

Reply:
but there is no worker, or is there grpc worker

Reply:
... you are right, I should get sleep